### PR TITLE
Add 'no-discard' option to formatting methods

### PR DESCRIPTION
**Pull request by mvollmer**
It unfortunately seems to be necessary to use this option in certain
situations.  See https://bugzilla.redhat.com/show_bug.cgi?id=1516041
for example.

### CODE REVIEW

File: src/udiskslinuxblock.c [hunk 1]
```diff
@@ -2656,6 +2656,27 @@ add_blocksize (gchar        **command,
...
+build_command (const gchar *template,
+               const gchar *device,
+               const gchar *label,
+               const gchar *options,
+               GError     **error)
+{
+  gchar *tmp, *tmp2, *command;
+  tmp = udisks_daemon_util_subst_str_and_escape (template, "$DEVICE", device);
+  tmp2 = udisks_daemon_util_subst_str_and_escape (tmp, "$LABEL", label != NULL ? label : "");
+  command = udisks_daemon_util_subst_str (tmp2, "$OPTIONS", options != NULL ? options : "");
```
Comment (vpodzime) [inline src/udiskslinuxblock.c]: This is ugly, but I don't see a better way to do it. Maybe there's some lightweight templating library we should switch to? Anyway, that's not the target here.

**Comment by vpodzime** [comment]
Jenkins, this is ok to test.

File: src/udiskslinuxblock.c (hunk 1 above)
Comment (mvollmer) [reply src/udiskslinuxblock.c]: asprintf?

**Comment by mvollmer** [comment]
> Looks good to me, but it would be nice if you modified the DBus tests that are already trying to create file systems to also try it with the new option.

Done.  Is there a way to check in `integration-tests` whether the expected option is actually passed to mkfs.ext4, for example?

File: src/udiskslinuxblock.c (hunk 1 above)
Comment (vpodzime) [reply src/udiskslinuxblock.c]: > asprintf?

That's not really the same thing as replacing values in templates, but it looks to me like the ``$something`` syntax here is an overkill anyway. Thanks for the tip!

**Comment by vpodzime** [comment]
> Done. Is there a way to check in integration-tests whether the expected option is actually passed to mkfs.ext4, for example?

None that I know of, unfortunately. And I'm afraid testing this in some other way would be too complicated to be worth it.

**Comment by mvollmer** [comment]
Thanks!

**Pull request merged**